src/misorders.ts

```diff
@@ -117,26 +147,94 @@ function getValidPathsForCoastalFleet(
 }
 
+function getValidPathsFromOrigin(
+  origin: string[],
+  isArmy: boolean
+): Record<string, number> {
+  if (!origin[1]) {
+    return getValidPathsWhenCoastsAreIrrelevant(origin[0], isArmy);
+  }
+  return getValidPathsForCoastalFleet(origin[0], origin[1] === "/sc");
+}
+
 function checkMoveOrderForMisorder(moveOrder: RegExpMatchArray): void {
   const isArmy: boolean = moveOrder[0][0] === "A";
   const origin: string[] = moveOrder.slice(1, 3);
   const destination: string = [moveOrder[3], moveOrder[4]]
     .join("")
     .replace("/", "");
-  let validPaths: Record<string, number>;
 
   if (isArmy && isOrderBetweenCoastalTerritories(origin[0], destination)) {
     return;
   }
 
-  if (!origin[1]) {
-    validPaths = getValidPathsWhenCoastsAreIrrelevant(origin[0], isArmy);
-  } else {
-    validPaths = getValidPathsForCoastalFleet(origin[0], origin[1] === "/sc");
-  }
+  const validPaths = getValidPathsFromOrigin(origin, isArmy);
 
   if (!(destination in validPaths)) {
-    const pathsOfMisorder = generatePathsOfMisorder(moveOrder);
-    colorMisorderRed(pathsOfMisorder);
+    const pathsOfMisorder = generatePathsOfMoveMisorder(moveOrder);
+    colorMoveMisorderRed(pathsOfMisorder);
+  }
+}
+
+function generatePathOfSupportHoldMisorder(
+  supportOrder: RegExpMatchArray
+): string {
+  const centerOfSupportedUnit = getCenterOfOriginFromOriginRegexArray(
+    supportOrder.slice(3, 5)
+  );
+
+  const centerOfSupportingUnit = getCenterOfOriginFromOriginRegexArray(
+    supportOrder.slice(1, 3)
+  );
+
+  const offset = Math.atan2(
+    centerOfSupportedUnit.y - centerOfSupportingUnit.y,
+    centerOfSupportedUnit.x - centerOfSupportingUnit.x
+  );
+
+  return [
+    "M",
+    centerOfSupportingUnit.x + 10 * Math.cos(offset),
+    ",",
+    centerOfSupportingUnit.y + 10 * Math.sin(offset),
+    "L",
+    centerOfSupportedUnit.x - 10 * Math.cos(offset),
+    ",",
+    centerOfSupportedUnit.y - 10 * Math.sin(offset),
+  ].join("");
+}
+
+function colorSupportHoldMisorderRed(misorderPath: string): void {
+  d3.select("svg")
+    .selectAll("path")
+    .filter(function () {
+      return d3.select(this).attr("d") == misorderPath;
+    })
+    .attr("stroke", "#ff0000");
+}
+
+function checkSupportHoldOrderForMisorder(
+  supportOrder: RegExpMatchArray
+): void {
+  const isArmy: boolean = supportOrder[0][0] === "A";
+  const origin: string[] = supportOrder.slice(1, 3);
+  const destination: string = [supportOrder[3], supportOrder[4]]
+    .join("")
+    .replace("/", "");
+
+  const validPaths = getValidPathsFromOrigin(origin, isArmy);
+
+  if (!(destination.substring(0, 3) in validPaths)) {
+    const misorderPath = generatePathOfSupportHoldMisorder(supportOrder);
+    colorSupportHoldMisorderRed(misorderPath);
+  }
+}
+
+function checkSupportOrderForMisorder(supportOrder: RegExpMatchArray): void {
+  const isSupportMove: boolean = !!supportOrder[5];
+
+  if (isSupportMove) {
```

Comment:
The more idiomatic way to do this would be to do return in the first condition, then flatten the else.

```
if (isSupportMove) {
  return;
}
checkSupportHoldOrderForMisorder(supportOrder);
}